source/mir/las/sum.d

```diff
@@ -1822,16 +1828,20 @@ nothrow unittest
         version(X86)
         {
             assert(summator.isNaN() == r.isNaN());
-            assert(summator.isFinite() == r.isFinite() || r == -double.max && s == -double.infinity || r == double.max && s == double.infinity);
-            assert(summator.isInfinity() == r.isInfinity() || r == -double.max && s == -double.infinity || r == double.max && s == double.infinity);
-            assert(nextDown(s) <= r && r <= nextUp(s) || s.isNaN && r.isNaN);
+            assert(summator.isFinite() == r.isFinite()
+                 || (r == -double.max && s == -double.infinity)
+                 || (r == double.max && s == double.infinity));
+            assert(summator.isInfinity() == r.isInfinity()
+                 || (r == -double.max && s == -double.infinity)
+                 || (r == double.max && s == double.infinity));
+            assert((nextDown(s) <= r && r <= nextUp(s)) || (s.isNaN && r.isNaN));
```

Comment:
Are u sure? it does make the code better to read. The idea is to avoid confusing expressions:

```
if (a && b || c) // bad
if (a && (b || c)) // good
```

It checks is if there are multiple `||` and `&&` in one expression,
so `1 && 2 && 3` will not trigger it.

It is called ["logical precedence check"](https://github.com/Hackerpilot/Dscanner/blob/master/src/analysis/logic_precedence.d)